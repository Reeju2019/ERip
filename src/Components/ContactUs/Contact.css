.contact {
  background-color: #f5f6fa;
  padding: 50px;
}

.contact-no {
  color: #194abb;
  font-weight: bolder;
}

.contact-time {
  color: gray;
  font-size: 10px;
}

.address {
  font-size: 13px !important;
}

.left-column {
  display: flex;
  flex-direction: column;
}

.form-footer {
  color: #0e62cb;
  font-size: 10px;
  margin-top: 15px;
  font-weight: bold;
  cursor: pointer;
}
