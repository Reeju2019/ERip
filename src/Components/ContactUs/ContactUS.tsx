import React from 'react'
import { Button, Col, Container, Form, Row } from 'react-bootstrap'
import './Contact.css'
import ContactPageData from '../../Data/ContactPageData.mock.json'
import Header from '../Home/Header/Header'

const ContactUs: React.FunctionComponent = () => {
  return (
    <>
      <Header />
      <Container fluid className='contact'>
        <Container>
          <Row>
            <Col className='col-12'>
              <div>
                <h4 className='mb-lg-4 mt-3 mb-3 d-flex justify-content-center'>
                  {ContactPageData.contact_title}
                </h4>
              </div>
            </Col>
          </Row>
          <Row className='d-flex'>
            <Col className='col-12 col-md-4 bg-white p-4'>
              <div className='left-column'>
                <h4 className='lg-4 fs-5 d-flex justify-content-left'>
                  {ContactPageData.contact_left_title}
                </h4>
                <div className='col-12'>
                  <hr className='mt-0' />
                </div>
                <p className='fw-bolder m-0 text-muted d-flex justify-content-left'>
                  {ContactPageData.contact_left_mid}
                </p>
                <div className='address text-muted m-0'>
                  <p className='m-0 d-flex justify-content-left'>
                    {ContactPageData.contact_left_l1}
                  </p>
                  <p className='m-0 d-flex justify-content-left'>
                    {ContactPageData.contact_left_l2}
                  </p>
                  <p className='m-0 d-flex justify-content-left'>
                    {ContactPageData.contact_left_l3}
                  </p>
                  <p className='m-0 d-flex justify-content-left text-decoration-none'>
                    {ContactPageData.contact_left_l4}
                  </p>
                </div>
                <a
                  className='d-flex justify-content-left text-decoration-none'
                  rel='noopener noreferrer'
                  href='https://www.google.com/maps/place/ERIP,+1709,+1st+Main+Rd,+Austin+Town,+Neelasandra,+Bengaluru,+Karnataka+560047/@12.9582971,77.6152016,17z/data=!4m5!3m4!1s0x3bae15605263ebe3:0x4067d9247ec36adf!8m2!3d12.9582971!4d77.6152016?hl=en-IN&gl=in'
                  target='_blank'
                >
                  {ContactPageData.get_direction}
                </a>
                <div className='col-12'>
                  <hr className='mt-10' />
                </div>
                <p className='d-flex justify-content-left m-0 text-muted'>
                  {ContactPageData.helpline_number}
                </p>
                <p className='mt-2 fs-6 '>
                  <a href='tel:080-47184455' className='d-flex justify-content-left'>
                    <span className='contact-no m-0'>{ContactPageData.helpline_no}</span>
                  </a>
                  <span className='contact-time d-flex justify-content-left m-0'>
                    {ContactPageData.time}
                  </span>
                </p>
              </div>
            </Col>
            <Col className='col-12 col-md-4'>
              <iframe
                src='https://www.google.com/maps/embed?pb=!1m14!1m8!1m3!1d3888.211763521468!2d77.6152016!3d12.9582971!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x3bae15605263ebe3%3A0x4067d9247ec36adf!2sERIP!5e0!3m2!1sen!2sin!4v1659006780823!5m2!1sen!2sin'
                width='100%'
                height='100%'
                // style='border:0;'
                // allowfullscreen=''
                loading='lazy'
                // referrerpolicy='no-referrer-when-downgrade'
              ></iframe>
            </Col>
            <Col className='col-12 col-md-4 bg-white p-4'>
              <Form>
                <h4 className='lg-4 fs-5 d-flex justify-content-left'>
                  {ContactPageData.contact_right_title}
                </h4>
                <div className='col0'>
                  <hr className='mt-10' />
                </div>
                <Form.Text className='text-muted d-flex justify-content-left'>
                  {ContactPageData.contact_right_para}
                </Form.Text>
                <Form.Group className='mt-3' controlId='formBasicName'>
                  <Form.Control type='text' placeholder='Name' />
                </Form.Group>
                <Form.Group className='mt-3' controlId='formBasicMobile'>
                  <Form.Control type='' placeholder='Mobile' />
                </Form.Group>
                <Form.Group className='mt-3' controlId='formBasicEmail'>
                  <Form.Control type='email' placeholder='Email' />
                </Form.Group>
                <Form.Group className='mt-3' controlId='formBasicMessage'>
                  <Form.Control as='textarea' rows={3} placeholder='Message' />
                </Form.Group>
                <Button className='mt-4'>Request a Callback</Button>
                <p className='form-footer'>{ContactPageData.connect_us}</p>
              </Form>
            </Col>
          </Row>
        </Container>
      </Container>
    </>
  )
}

export default ContactUs
