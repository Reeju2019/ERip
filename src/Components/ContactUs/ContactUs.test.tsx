import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import ContactUs from './ContactUS'
import { MemoryRouter as Router } from 'react-router-dom'

describe('App rendering specification', () => {
  test('contact us', () => {
    render(
      <Router>
        <ContactUs />
      </Router>,
    )
  })
  it('SnapShot testing', () => {
    const component = renderer.create(
      <Router>
        <ContactUs />
      </Router>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('render without crashing', () => {
    const div = document.createElement('div')
    render(
      <Router>
        <ContactUs />
      </Router>,
    )
    ReactDOM.unmountComponentAtNode(div)
  })
})

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
}))
