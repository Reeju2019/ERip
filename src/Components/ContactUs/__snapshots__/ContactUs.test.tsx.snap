// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering specification SnapShot testing 1`] = `
Array [
  <div
    className=" py-1 m-0 header-comp container-fluid"
  >
    <div
      className="container"
    >
      <div
        className="Navbar row"
      >
        <div
          className="col-1 align-item-center align-self-center col"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="logo"
              className="col-2 Logo align-self-center"
              src="/static/image/logo_erip.png"
            />
          </a>
        </div>
        <div
          className="modal-search d-flex col-6 col"
        >
          <button
            className="btn modal-btn rounded-0 bg-transparent text-muted fw-bold btn btn-primary"
            data-target="#myModal"
            data-toggle="modal"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <i
              className="fas fa-map-marker-alt"
            />
            Bengaluru
            <i
              className="fa-solid fa-caret-down"
            />
          </button>
          <div
            className="search_btn d-flex input-group"
          >
            <div
              className="align-self-center m-3"
            >
              <i
                className="fas fa-search"
              />
            </div>
            <input
              aria-describedby="search-addon"
              aria-label="Search"
              className="form-control rounded-0 border-0 form-control"
              id="search-border"
              placeholder="Search for a services"
              type="search"
            />
          </div>
        </div>
        <div
          className="col-5 align-self-center col"
        >
          <div
            className="modal-search d-flex justify-content-end align-self-center nav"
            onKeyDown={[Function]}
          >
            <a
              className="text-muted fw-bold nav-link"
              data-rr-ui-event-key="/"
              disabled={false}
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
            <a
              className="text-muted fw-bold nav-link"
              data-rr-ui-event-key="/contact"
              disabled={false}
              href="/contact"
              onClick={[Function]}
            >
              Contact Us
            </a>
            <a
              className="d-flex flex-row text-muted fw-bold nav-link"
              data-rr-ui-event-key={null}
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <i
                className="fa-regular fa-user"
              />
              Login
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="contact container-fluid"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12 col"
        >
          <div>
            <h4
              className="mb-lg-4 mt-3 mb-3 d-flex justify-content-center"
            >
              We'd Love to hear from You!
            </h4>
          </div>
        </div>
      </div>
      <div
        className="d-flex row"
      >
        <div
          className="col-12 col-md-4 bg-white p-4 col"
        >
          <div
            className="left-column"
          >
            <h4
              className="lg-4 fs-5 d-flex justify-content-left"
            >
              Get in Touch
            </h4>
            <div
              className="col-12"
            >
              <hr
                className="mt-0"
              />
            </div>
            <p
              className="fw-bolder m-0 text-muted d-flex justify-content-left"
            >
              Head Office
            </p>
            <div
              className="address text-muted m-0"
            >
              <p
                className="m-0 d-flex justify-content-left"
              >
                ERIP, 1709, 1st Main Rd,
              </p>
              <p
                className="m-0 d-flex justify-content-left"
              >
                Austin Town, Neelasandra,
              </p>
              <p
                className="m-0 d-flex justify-content-left"
              >
                Bengaluru, Karnataka 560047
              </p>
              <p
                className="m-0 d-flex justify-content-left text-decoration-none"
              >
                Tel - 080-47184455
              </p>
            </div>
            <a
              className="d-flex justify-content-left text-decoration-none"
              href="https://www.google.com/maps/place/ERIP,+1709,+1st+Main+Rd,+Austin+Town,+Neelasandra,+Bengaluru,+Karnataka+560047/@12.9582971,77.6152016,17z/data=!4m5!3m4!1s0x3bae15605263ebe3:0x4067d9247ec36adf!8m2!3d12.9582971!4d77.6152016?hl=en-IN&gl=in"
              rel="noopener noreferrer"
              target="_blank"
            >
              GET DIRECTION
            </a>
            <div
              className="col-12"
            >
              <hr
                className="mt-10"
              />
            </div>
            <p
              className="d-flex justify-content-left m-0 text-muted"
            >
              Helpline Number
            </p>
            <p
              className="mt-2 fs-6 "
            >
              <a
                className="d-flex justify-content-left"
                href="tel:080-47184455"
              >
                <span
                  className="contact-no m-0"
                >
                  080-47184455
                </span>
              </a>
              <span
                className="contact-time d-flex justify-content-left m-0"
              >
                *10 am - 7 pm (All Days)
              </span>
            </p>
          </div>
        </div>
        <div
          className="col-12 col-md-4 col"
        >
          <iframe
            height="100%"
            loading="lazy"
            src="https://www.google.com/maps/embed?pb=!1m14!1m8!1m3!1d3888.211763521468!2d77.6152016!3d12.9582971!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x3bae15605263ebe3%3A0x4067d9247ec36adf!2sERIP!5e0!3m2!1sen!2sin!4v1659006780823!5m2!1sen!2sin"
            width="100%"
          />
        </div>
        <div
          className="col-12 col-md-4 bg-white p-4 col"
        >
          <form
            className=""
          >
            <h4
              className="lg-4 fs-5 d-flex justify-content-left"
            >
              Need Help?
            </h4>
            <div
              className="col0"
            >
              <hr
                className="mt-10"
              />
            </div>
            <small
              className="text-muted d-flex justify-content-left form-text"
            >
              Fill the form below to get a callback.
            </small>
            <div
              className="mt-3"
            >
              <input
                className="form-control"
                id="formBasicName"
                placeholder="Name"
                type="text"
              />
            </div>
            <div
              className="mt-3"
            >
              <input
                className="form-control"
                id="formBasicMobile"
                placeholder="Mobile"
                type=""
              />
            </div>
            <div
              className="mt-3"
            >
              <input
                className="form-control"
                id="formBasicEmail"
                placeholder="Email"
                type="email"
              />
            </div>
            <div
              className="mt-3"
            >
              <textarea
                className="form-control"
                id="formBasicMessage"
                placeholder="Message"
                rows={3}
              />
            </div>
            <button
              className="mt-4 btn btn-primary"
              disabled={false}
              type="button"
            >
              Request a Callback
            </button>
            <p
              className="form-footer"
            >
              Or Connect Us on Whatsapp
            </p>
          </form>
        </div>
      </div>
    </div>
  </div>,
]
`;
